llvm/include/llvm/ADT/DenseMapInfo.h

```diff
@@ -137,7 +137,10 @@ template<> struct DenseMapInfo<unsigned long> {
   static inline unsigned long getTombstoneKey() { return ~0UL - 1L; }
 
   static unsigned getHashValue(const unsigned long& Val) {
-    return (unsigned)(Val * 37UL);
+    if constexpr (sizeof(Val) == 4)
+      return DenseMapInfo<unsigned>::getHashValue(Val);
+    else
+      return detail::combineHashValue(Val >> 32, Val);
```

Comment:
@ChuanqiXu9  To silence MSVC warnings, could we change this to:
```
return detail::combineHashValue(Val >> (4 * sizeof(Val)), Val);
```

Reply:
Thanks. You can land that directly to make the build green. But I don't understand in what cases it is a problem. I think I've already handled the case for its size is 4. I don't feel it is possible that `sizeof(Val)` may evaluate to `2` here.